server/src/modules/stats/board/board.stats.service.ts

```diff
@@ -300,4 +300,95 @@ export class BoardStatsService implements OnModuleInit, OnApplicationShutdown {
 
     return this.convertTimesSpentMapToTimeSpentDataArray(timesSpentMap);
   }
+
+  async getBoardActionsOverTime(
+    boardId: string,
+    startDate: Date,
+    endDate: Date,
+    aggregationIntervalMinutes: number,
+  ): Promise<NumericalTimelineChartData[]> {
+    if (startDate.toString() === 'Invalid Date') startDate = new Date(0);
+    if (endDate.toString() === 'Invalid Date' || endDate > new Date())
+      endDate = new Date();
+
+    if (startDate >= endDate) {
+      throw new HttpException('Invalid dates', HttpStatus.BAD_REQUEST);
+    }
+
+    const boardStats = await this.boardStatsModel.findOne({
+      boardId,
+      $and: [
+        { 'actionTimeline.timestamp': { $gte: startDate } },
+        { 'actionTimeline.timestamp': { $lte: endDate } },
+      ],
+    });
+
+    if (!boardStats || !boardStats.actionTimeline) {
+      return [];
+    }
+
+    const boardActionsOverTimeMap = this.getBoardActionsOverTimeMap(
+      boardStats.actionTimeline,
+      startDate.getTime(),
+      endDate.getTime(),
+      aggregationIntervalMinutes * 60 * 1000,
+    );
+
+    return this.convertBoardActionsOverToChartDataArray(
+      boardActionsOverTimeMap,
+    );
+  }
+
+  getBoardActionsOverTimeMap(
+    actionTimeline: [
```

Comment:
Please add interface

Reply:
I changed the parameter to BoardStats snd retrieve actionTimeline later